src/postgres-util/src/lib.rs

```diff
@@ -207,15 +206,139 @@ pub async fn drop_replication_slots(config: Config, slots: &[&str]) -> Result<()
     Ok(())
 }
 
-/// Starts a replication connection to the upstream database
-pub async fn connect_replication(mut config: Config) -> Result<Client, anyhow::Error> {
-    let tls = make_tls(&config)?;
-    let (client, connection) = config
-        .replication_mode(ReplicationMode::Logical)
-        .connect_timeout(Duration::from_secs(30))
-        .keepalives_idle(Duration::from_secs(10 * 60))
-        .connect(tls)
-        .await?;
-    task::spawn(|| "postgres_connect_replication", connection);
-    Ok(client)
+#[derive(PartialEq, Clone)]
+pub enum SshTunnelConfig {
+    Direct,
+    Tunnel {
+        host: String,
+        port: u16,
+        user: String,
+        public_key: String,
+        private_key: String,
```

Comment:
its really weird that the ssh library takes `&str` for the private and public key, no `&[u8]`...in any case, can we store a https://docs.rs/ssh-key/0.4.2/ssh_key/private/struct.Ed25519Keypair.html instead and then convert to `str` as we convert?

Reply:
That's the logic that is happening in `ore::ssh_key` when we generate the keys, but since we never need to process it (other than passing it as an `&str` here) and it's a string anyway, I just kept it as a `String` here after reading it from the secrets store. Do you think the additional conversion to the `ssh-key` would help here?